src/arch/helperrvv.h

```diff
@@ -72,30 +72,30 @@
 #error Cannot simultaneously define ENABLE_RVV_SP and ENABLE_RVV_DP
 #endif
 
-#ifdef ENABLE_RVV_SP
-// Types that conflict with ENABLE_RVV_DP
 #if defined(ENABLE_RVVM1) || defined(ENABLE_RVVM1NOFMA)
-typedef vuint64m2_t vmask;
-typedef vbool32_t vopmask;
+typedef vuint64m2_t rvv_sp_vmask;
+typedef vuint64m1_t rvv_dp_vmask;
+typedef vbool32_t rvv_sp_vopmask;
+typedef vbool64_t rvv_dp_vopmask;
 #elif defined(ENABLE_RVVM2) || defined(ENABLE_RVVM2NOFMA)
-typedef vuint64m4_t vmask;
-typedef vbool16_t vopmask;
+typedef vuint64m4_t rvv_sp_vmask;
+typedef vuint64m2_t rvv_dp_vmask;
+typedef vbool16_t rvv_sp_vopmask;
+typedef vbool32_t rvv_dp_vopmask;
```

Comment:
I needed these extra types because I could not use the `ENABLE_RVV_SP` macro to disable some functions which used them internally, because they were used externally.